src/ted2zim/scraper.py

```diff
@@ -1330,6 +1336,46 @@ def remove_failed_topics_and_check_extraction(self, failed_topics):
                     "topic(s) and/or language code supplied to --languages"
                 )
             raise ValueError("Wrong topic(s) were supplied. No videos found")
+        
+    def extract_videos_from_links(self, links):
+        """extracts metadata for all videos specified by links
+        
+        calls extract_info_from_video_page on all links to get this data
+        """
+        logger.debug(f"Extracting videos from {len(links)} links")
+        
+        for url in links:
+            url = url.strip()
+            # Ensure the URL is a valid TED talk URL
+            if not url.startswith(BASE_URL) and not "ted.com/talks" in url:
+                logger.warning(f"Skipping invalid TED talk URL: {url}")
+                continue
+            
+            logger.debug(f"Processing link: {url}")
+            json_data = self.extract_info_from_video_page(url)
+            
+            # Skip if we couldn't retrieve the data
+            if json_data is None:
+                continue
+            
+            # Process the video data
+            if self.update_videos_list_from_info(json_data):
+                # Try to get the talk in other languages if source_languages specified
+                if self.source_languages:
+                    urls = self.generate_urls_for_other_languages(
+                        url, self.source_languages
+                    )
+                    for url in urls:
+                        json_data = self.extract_info_from_video_page(url)
+                        if json_data:
+                            self.update_videos_list_from_info(json_data)
+            
+            self.already_visited.add(urllib.parse.urlparse(url).path)
```

Comment:
Is this really necessary ? I thought so far this is used only to not add twice the same video which might be in multiple topics ; I don't think we have to handle the case where someone pass the same link twice ...